src/main/java/frc/robot/filters/AutoYawFilter.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.filters;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.interfaces.Gyro;
+
+/**
+ *
+ * 
+ * @author aheitkamp
+ */
+public class AutoYawFilter extends Filter {
+    private Gyro navX;
+    private PIDController PID;
+
+    public AutoYawFilter(Gyro NavX) {
+        navX = NavX;
+
+        PID = new PIDController(0.095, 0.15, 0.0085);
+        PID.setTolerance(2);
+        PID.enableContinuousInput(-180, 180);
+    }
+
+    public void filter(DriveInput DI) {
+        if (!enable) {
+            PID.reset();
+            return;
+        }
+
+        if (Math.abs(DI.getForward()) < 0.1 && Math.abs(DI.getRight()) < 0.1) {
+            PID.reset();
+            return;
+        }
+
+        if (DI.getOverrideAutoYaw()) {
```

Comment:
i think there are too many branches here that do the same thing: PID.reset() then return. 
this should be factored out